controller/konnect/ops/ops_kongupstream.go

```diff
@@ -85,6 +87,59 @@ func deleteUpstream(
 	return nil
 }
 
+func adoptUpstream(
+	ctx context.Context,
+	sdk sdkops.UpstreamsSDK,
+	upstream *configurationv1alpha1.KongUpstream,
+) error {
+	cpID := upstream.GetControlPlaneID()
+	adoptOptions := upstream.Spec.Adopt
+	konnectID := adoptOptions.Konnect.ID
+	if cpID == "" {
+		return errors.New("No Control Plane ID")
+	}
+	resp, err := sdk.GetUpstream(ctx, konnectID, cpID)
+	if err != nil {
+		return KonnectEntityAdoptionFetchError{
+			KonnectID: konnectID,
+			Err:       err,
+		}
+	}
+	uidTag, hasUIDTag := findUIDTag(resp.Upstream.Tags)
+	if hasUIDTag && extractUIDFromTag(uidTag) != string(upstream.UID) {
+		return KonnectEntityAdoptionUIDTagConflictError{
+			KonnectID:    konnectID,
+			ActualUIDTag: extractUIDFromTag(uidTag),
+		}
+	}
+
+	adoptMode := adoptOptions.Mode
+	if adoptMode == "" {
+		adoptMode = commonv1alpha1.AdoptModeOverride
+	}
+	switch adoptOptions.Mode {
```

Comment:
```suggestion
	switch adoptMode {
```

We have changed the default value for it.

Reply:
Fixed